### DISCUSSION TITLE
ClawPDF COM - ShowProgress not working

### DISCUSSION THREAD

**Discussion by mpreis1976** [clawsoftware/clawPDF#147] [Q&A]
I do convert several print-outs from a CAD-application using ClawPDFs COM interface which works quite good.

Here is my (AutoIT) Code how it's done:

````
$oPDFCreatorQueue = ObjCreate('ClawPDF.JobQueue')
$oPDFCreatorQueue.Initialize()

If  $oPDFCreatorQueue.WaitForJobs($ANZAHL_PL, 9999) Then


	Local $oJob = Null
	If $oPDFCreatorQueue.Count = 0 Then
		$oPDFCreatorQueue.ReleaseCom()
		$oPDFCreatorQueue = Null
		exit
	Else
		$oPDFCreatorQueue.MergeAllJobs
		LOGGING("Merge" & @CRLF)
		$oJob = $oPDFCreatorQueue.NextJob
		LOGGING("NextJob" & @CRLF)

		$oJob.SetProfileByGuid('DefaultGuid')
		$oJob.SetProfileSetting('OpenViewer', 'False')
		$oJob.SetProfileSetting('ShowProgress', 'True')
		$oJob.SetProfileSetting('PdfSettings.PageOrientation', 'Landscape')
		LOGGING("Settings" & @CRLF)

		$pJobInfo = $oJob.PrintJobInfo
		$PJName = $pJobInfo.PrintJobName

		LOGGING($DATEINAME & @CRLF)

		$oJob.ConvertTo($DATEINAME)
		LOGGING("Convert" & @CRLF)

	EndIf

EndIf

$oPDFCreatorQueue.ReleaseCom()
$oPDFCreatorQueue = Null

LOGGING("ReleaseCom" & @CRLF)
````

My problem is that the conversion of a multipage-PDF can take up to 2 minutes where the main application is idle and the user could possible think everything is set and done.

Therefore I added the line `$oJob.SetProfileSetting('ShowProgress', 'True')` to show the progress of the file creation.

Unfortunately the added line of code did not bring any effect. There is no progress information, no window, nothing ...

Is there anything wrong with my code or do I have to add something more to see a progress indicator?

**Comment by toledano** [comment]
This run perfectly with VB calling COM interface
printJob.SetProfileSetting "ShowProgress", "True"                                                ' Bool    If true, displays a progress window during conversion.

!!! TAKE CARE - CONSUMING RESOURCE and TIME. It not possible to trigger progession... Not usefull
AUTOCAD seems using same VBA language, but syntax is different ?
